crates/mpz-garble/src/protocol/deap/mod.rs

```diff
@@ -139,6 +139,86 @@ impl DEAP {
         self.state.lock().unwrap()
     }
 
+    /// Commits the provided input values.
+    pub async fn commit<Ctx, OTS, OTR>(
+        &self,
+        ctx: &mut Ctx,
+        inputs: &[ValueRef],
+        ot_send: &mut OTS,
+        ot_recv: &mut OTR,
+    ) -> Result<(), DEAPError>
+    where
+        Ctx: Context,
+        OTS: OTSendEncoding<Ctx> + Send,
+        OTR: OTReceiveEncoding<Ctx> + Send,
+    {
+        let assigned = self.state().memory.drain_assigned(inputs);
+        match self.role {
+            Role::Leader => {
+                try_join!(
+                    ctx,
+                    self.gen
+                        .setup_assigned_values(ctx, &assigned, ot_send)
+                        .map_err(DEAPError::from),
+                    self.ev
+                        .setup_assigned_values(ctx, &assigned, ot_recv)
+                        .map_err(DEAPError::from)
+                )??;
+            }
+            Role::Follower => {
+                try_join!(
+                    ctx,
+                    self.ev
+                        .setup_assigned_values(ctx, &assigned, ot_recv)
+                        .map_err(DEAPError::from),
+                    self.gen
+                        .setup_assigned_values(ctx, &assigned, ot_send)
+                        .map_err(DEAPError::from)
+                )??;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Commits the provided values for proving.
+    pub async fn commit_prove<Ctx, OTR>(
+        &self,
+        ctx: &mut Ctx,
+        values: &[ValueRef],
+        ot_recv: &mut OTR,
+    ) -> Result<(), DEAPError>
+    where
+        Ctx: Context,
+        OTR: OTReceiveEncoding<Ctx> + Send,
+    {
+        let assigned = self.state().memory.drain_assigned(values);
+        self.ev
+            .setup_assigned_values(ctx, &assigned, ot_recv)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Commits the provided values for verifying.
```

Comment:
Reading the comment may seem confusing since the verifier is not actually committing to anything, instead maybe 
```suggestion
    /// Enables the prover to commit the provided values for verifying.
```

Reply:
Think of it more like a "database commit" instead of cryptographic commitment. The two parties have a shared virtual machine and are jointly committing values to its memory.

Reply:
Ideally, we should avoid leaking the VM impl details into the DEAP protocol API, if possible.

The person who follows the DEAP spec doc will be confused by the sudden overload of the term "commit", imo. To avoid the confusion, we need to explain the term "commit" from the cryptographic commitment pov.